sources/console.c

```diff
@@ -793,6 +798,127 @@ od_console_set(od_client_t *client, machine_msg_t *stream)
 	return 0;
 }
 
+static inline int
+od_console_route_set_storage_state_cb(od_route_t *route, void **argv)
+{
+    char *storage_name = argv[0];
+    od_client_t *client = argv[1];
+    od_rule_storage_state_t *state = (od_rule_storage_state_t*)argv[2];
+    od_instance_t *instance = client->global->instance;
+    bool pause_all = strcmp(storage_name, "") == 0;
+
+	if (route->rule->obsolete)
+		return 0;
+    if (route->rule->storage->storage_type != OD_RULE_STORAGE_REMOTE)
+        return 0;
+
+    if (!(pause_all || strcmp(route->rule->storage->name,storage_name) == 0))
+        return 0;
+
+    route->rule->storage->state = *state;
+
+    return pause_all ? 0 : 1;
+}
+
+static inline int
+od_console_route_wait_paused_cb(od_route_t *route, void **argv)
+{
+    char *storage_name = argv[0];
+    od_client_t *client = argv[1];
+    od_instance_t *instance = client->global->instance;
+    bool pause_all = strcmp(storage_name, "") == 0;
+    for(;;)
+    {
+        if (route->rule->obsolete)
+            return 0;
+        if (route->rule->storage->storage_type != OD_RULE_STORAGE_REMOTE)
+            return 0;
+
+        if (!(pause_all || strcmp(route->rule->storage->name,storage_name) == 0))
+            return 0;
+
+        route->rule->storage->state = OD_STORAGE_PAUSE;
+
+
+        od_route_lock(route);
+
+        bool stopped = false;
+        if(!route->server_pool.count_idle && !route->server_pool.count_active)
+            stopped = true;
+
+        od_route_unlock(route);
+
+        if(stopped)
+            return pause_all ? 0 : 1;
+        machine_sleep(100);
+    }
+}
+
+static inline int
+od_console_query_set_storage_state(od_client_t *client, machine_msg_t *stream,
+                                   od_parser_t *parser,
+                                   od_rule_storage_state_t state) {
+    machine_msg_t *msg;
+    od_instance_t *instance = client->global->instance;
+    od_router_t *router = client->global->router;
+
+    char *storage_name = "";
+    od_token_t token;
+    int rc = od_parser_next(parser, &token);
+
+    char *state_name = state == OD_STORAGE_PAUSE ? "PAUSED" : "RESUME";
+
+    if (rc == OD_PARSER_KEYWORD)
+    {
+        storage_name = token.value.string.pointer;
+        od_log(&instance->logger, "console", client, NULL,
+                "Making storage %.*s %s...", token.value.string.size, token.value.string.pointer, state_name);
+    }
+    else
+    {
+        od_log(&instance->logger, "console", client, NULL,
```

Comment:
During IRL discussion with @x4m we decided not to implement pause/resume for session pooling. So I am not sure what "for all storages" operation should do. Should it silently apply change only to storages with transaction pooling or report problems with session pooling storages or not work at all if any session pooling storage found?